sql/src/main/java/io/crate/memory/OnHeapMemoryManager.java

```diff
@@ -38,6 +38,8 @@ public OnHeapMemoryManager(IntConsumer accountBytes) {
     @Override
     public ByteBuf allocate(int capacity) {
         accountBytes.accept(capacity);
+        // TODO: should we hold onto the buffers and release them as well?
+        // contract would require it, but there is no resource behind it that must be released
```

Comment:
hm not sure if we should do that just to fullfil the contract without technical reasoning. I'd rather add a note here about that.